src/Confluent.Kafka/Impl/SafeKafkaHandle.cs

```diff
@@ -1596,6 +1596,56 @@ internal void DeleteGroups(IList<string> deleteGroups, DeleteGroupsOptions optio
             }
         }
 
+        internal void DeleteConsumerGroupOffsets(string group, IEnumerable<TopicPartitionOffset> partitions, DeleteConsumerGroupOffsetsOptions options, IntPtr resultQueuePtr, IntPtr completionSourcePtr)
+        {
+            ThrowIfHandleClosed();
+
+            options = options == null ? new DeleteConsumerGroupOffsetsOptions() : options;
+
+            IntPtr[] partitionsPtrs = new IntPtr[partitions.Count()];
+            IntPtr optionsPtr = IntPtr.Zero;
+            try
+            {
+                optionsPtr = Librdkafka.AdminOptions_new(handle, Librdkafka.AdminOp.DeleteConsumerGroupOffsets);
+                setOption_RequestTimeout(optionsPtr, options.RequestTimeout);
+                setOption_OperationTimeout(optionsPtr, options.OperationTimeout);
+                setOption_completionSource(optionsPtr, completionSourcePtr);
+
+                if (partitions.Where(tpo => tpo.Topic == null).Count() > 0)
+                {
+                    throw new ArgumentException("Cannot delete offsets because one or more topics were specified as null.");
+                }
+
+                IntPtr cOffsets = GetCTopicPartitionList(partitions);
+                if (cOffsets == IntPtr.Zero)
+                {
+                    throw new ArgumentNullException("Delete offsets partitions collection must not be null");
+                }
+
+                for (int i = 0; i < partitions.Count(); i++)
```

Comment:
this isn't what we want - have a re-think, can discuss if you want.

Reply:
Thanks for catching this. I have updated the interface to now receive an enumerable of `ConsumerGroupOffsets` and then iterating over it to create the `DeleteConsumerGroupOffsets` req.